job-simulation.rb

```diff
@@ -1,4 +1,47 @@
 require './Stack.rb'
 require './Queue.rb'
 
+class Company
+  MAX_EMPLOYEES = 6
 
+  attr_reader :waiting, :working
+
+  def initialize
+    @waiting = Queue.new
+    10.times { |i| wait! i+1 }
+
+    @working = Stack.new
+
+    end_of_the_month
+  end
+
+  def end_of_the_month
+    fire!
+    hire!
+  end
+
+  private
+
+  def hire!
+    until working.size >= MAX_EMPLOYEES
+      new_hire = waiting.dequeue
+      working.push new_hire
+    end
+  end
+
+  def fire!
+    return if working.empty?
+    number_to_fire.times do
+      fired_person = working.pop
+      wait! fired_person
```

Comment:
Same question here questioning the necessity of fired_person. Maybe we could say `wait! working.pop`, unless for some reason we don't want to do that? Maybe if working.pop returns nil somehow?